fe/fe-core/src/main/java/com/starrocks/transaction/TransactionStmtExecutor.java

```diff
@@ -0,0 +1,499 @@
+import com.starrocks.task.LoadEtlTask;
+import com.starrocks.thrift.TExplainLevel;
+import com.starrocks.thrift.TLoadJobType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.starrocks.common.ErrorCode.ERR_TXN_NOT_EXIST;
+
+public class TransactionStmtExecutor {
+    private static final Logger LOG = LogManager.getLogger(TransactionStmtExecutor.class);
+
+    public static void beginStmt(ConnectContext context, BeginStmt stmt) {
+        if (context.getExplicitTxnState() != null) {
+            //Repeated begin does not create a new transaction
+            ExplicitTxnState explicitTxnState = context.getExplicitTxnState();
+            String label = explicitTxnState.getTransactionState().getLabel();
+            long transactionId = explicitTxnState.getTransactionState().getTransactionId();
+            context.getState().setOk(0, 0, buildMessage(label, TransactionStatus.PREPARE, transactionId, -1));
+            return;
+        }
+
+        long transactionId = GlobalStateMgr.getCurrentState().getGlobalTransactionMgr()
+                .getTransactionIDGenerator().getNextTransactionId();
+        String label = DebugUtil.printId(context.getExecutionId());
+        TransactionState transactionState = new TransactionState(transactionId, label, null,
+                TransactionState.LoadJobSourceType.INSERT_STREAMING,
+                new TransactionState.TxnCoordinator(TransactionState.TxnSourceType.FE, FrontendOptions.getLocalHostAddress()),
+                context.getExecTimeout() * 1000L);
+
+        transactionState.setPrepareTime(System.currentTimeMillis());
+        transactionState.setWarehouseId(context.getCurrentWarehouseId());
+        boolean combinedTxnLog = LakeTableHelper.supportCombinedTxnLog(TransactionState.LoadJobSourceType.INSERT_STREAMING);
+        transactionState.setUseCombinedTxnLog(combinedTxnLog);
+
+        ExplicitTxnState explicitTxnState = new ExplicitTxnState();
+        explicitTxnState.setTransactionState(transactionState);
+        context.setExplicitTxnState(explicitTxnState);
+
+        context.getState().setOk(0, 0, buildMessage(label, TransactionStatus.PREPARE, transactionId, -1));
+    }
+
+    public static void loadData(Database database,
+                                Table targetTable,
+                                ExecPlan execPlan,
+                                DmlStmt dmlStmt,
+                                OriginStatement originStmt,
+                                ConnectContext context) {
+        Coordinator coordinator = new DefaultCoordinator.Factory().createInsertScheduler(
+                context, execPlan.getFragments(), execPlan.getScanNodes(), execPlan.getDescTbl().toThrift());
+        ExplicitTxnState explicitTxnState = context.getExplicitTxnState();
+        TransactionState transactionState = explicitTxnState.getTransactionState();
```

Comment:
no abstraction here, `ExplicitTxnState` sounds quite list it's a sub class of `TransactionState`.